src/ui/common/src/components/workflows/triggerSourceSelector.tsx

```diff
@@ -3,34 +3,40 @@ import FormControl from '@mui/material/FormControl';
 import MenuItem from '@mui/material/MenuItem';
 import Select from '@mui/material/Select';
 import Typography from '@mui/material/Typography';
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect, useState } from 'react';
 import { ListWorkflowSummary } from 'src/utils/workflows';
 
-import { RootState } from '../../stores/store';
-
 type Props = {
+  sourceId: string;
+  setSourceId: (string) => void;
   workflows: ListWorkflowSummary[];
 };
 
-export const TriggerSourceSelector: React.FC<Props> = ({ workflows }) => {
-  const workflow = useSelector((state: RootState) => state.workflowReducer);
+export const TriggerSourceSelector: React.FC<Props> = ({
+  sourceId,
+  setSourceId,
+  workflows,
+}) => {
+  const [selected, setSelected] = useState<ListWorkflowSummary>(
+    workflows.find((workflow) => workflow.id === sourceId)
+  );
 
-  const dag = workflow.selectedDag;
+  useEffect(() => {
+    if (!selected) {
+      return;
+    }
 
-  const filteredWorkflows = workflows.filter(
-    (workflow) => workflow.id === dag.metadata?.schedule.source_id
-  );
-  // TODO: ENG-2181 Add support for changing source trigger
-  const selected = filteredWorkflows[0].id;
+    setSourceId(selected.id);
+  }, [selected, setSourceId]);
 
   const getMenuItems = () => {
-    return filteredWorkflows.map((workflow) => {
+    return workflows.map((workflow) => {
       return (
+        //@ts-ignore
```

Comment:
@agiron123 I needed to add this tag here to use `value={workflow}`, but that's now causing a linter error. Unsure what the proper change should be.